exercise/submission_models.py

```diff
@@ -686,3 +710,31 @@ def _delete_file(sender, instance, **kwargs):
     """
     instance.file_object.delete(save=False)
 post_delete.connect(_delete_file, SubmittedFile)
+
+
+class PendingSubmission(models.Model):
+    submission = DefaultForeignKey('Submission',
+        verbose_name=_('LABEL_SUBMISSION'),
+        on_delete=models.CASCADE,
+    )
+    submission_time = models.DateTimeField(
+        verbose_name=_('LABEL_SUBMISSION_TIME'),
+        null=True,  # to make usage with get_or_create easier
+    )
+    num_retries = models.IntegerField(
+        verbose_name=_('LABEL_NUMBER_OF_RETRIES'),
+        default=0,
+    )
+
+    class Meta:
+        verbose_name = _('MODEL_NAME_PENDING_SUBMISSION')
+        verbose_name_plural = _('MODEL_NAME_PENDING_SUBMISSION_PLURAL')
+        constraints = [models.UniqueConstraint(fields=['submission'], name='unique_submission')]
```

Comment:
Yes, I tried that initially, but ran into some problems (don't remember what, since it was three months ago, maybe migrations related), but for some reason the foreign key approach appeared to work. However, now that I retried OneToOne, it works properly. Don't know what it was back then, but OneToOne is indeed how it should be.